portlet-parent/wicketstuff-portlet/src/main/java/org/apache/wicket/portlet/WicketPortlet.java

```diff
@@ -412,23 +412,7 @@ private void processMimeResponseRequest(String wicketURL, final PortletRequest r
 				// processing the request
 
 				String redirectLocation = responseState.getRedirectLocation();
-				String ajaxRedirectLocation = responseState.getAjaxRedirectLocation();
-				if (ajaxRedirectLocation != null) {

```

Comment:
@kkaravitis suggested to preserve the cookies before clearing the state and to re-apply them after that.

Reply:
@kkaravitis, @martin-g 
Do you think that there is a reason to clear the state at all? 
This is a redriect to any location a user wants. Inside the portal or outside it. 
A user might want to redirect to another site and add some authentication headers for example.

I understand that I found this from my cookies problem, but maybe I should rephrase it.
Unless you know of a reason for the the response state to be cleared..

Reply:
I insist on my first suggestion. I have doubts that the redirect occur if we remove all the logic of ajaxriderct url. Please see also my other comments on this commit.